File: source/src/main/java/org/rss/tools/mpl/reader/Reader.java
package org.rss.tools.mpl.reader;

import java.io.IOException;
import java.io.InputStream;

/**
 * Base of reader for multiple sources (implementations)
 * 
 * @author ricardo saturnino
 */
public interface Reader {

	InputStream read(String key) throws IOException;
}

File: source/src/main/kotlin/org/rss/tools/mpl/reader/Reader.kt
package org.rss.tools.mpl.reader

import java.io.IOException
import java.io.InputStream

/**
 * Base reader, allowing multiple sources (implementations)
 *
 * @author ricardo saturnino
 */
interface Reader {

    @Throws(IOException::class)
    fun read(key: String): InputStream
}
